python/semantic_kernel/connectors/ai/bedrock/services/bedrock_chat_completion.py

```diff
@@ -0,0 +1,406 @@
+        """
+        try:
+            bedrock_settings = BedrockSettings.create(
+                chat_model_id=model_id,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as e:
+            raise ServiceInitializationError("Failed to initialize the Amazon Bedrock Chat Completion Service.") from e
+
+        if bedrock_settings.chat_model_id is None:
+            raise ServiceInitializationError("The Amazon Bedrock Chat Model ID is missing.")
+
+        super().__init__(
+            ai_model_id=bedrock_settings.chat_model_id,
+            service_id=service_id or bedrock_settings.chat_model_id,
+            bedrock_runtime_client=runtime_client or boto3.client("bedrock-runtime"),
+            bedrock_client=client or boto3.client("bedrock"),
+        )
+
+    # region Overriding base class methods
+
+    # Override from AIServiceClientBase
+    @override
+    def get_prompt_execution_settings_class(self) -> type["PromptExecutionSettings"]:
+        return BedrockChatPromptExecutionSettings
+
+    @override
+    @trace_chat_completion(BedrockBase.MODEL_PROVIDER_NAME)
+    async def _inner_get_chat_message_contents(
+        self,
+        chat_history: "ChatHistory",
+        settings: "PromptExecutionSettings",
+    ) -> list["ChatMessageContent"]:
+        if not isinstance(settings, BedrockChatPromptExecutionSettings):
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, BedrockChatPromptExecutionSettings)  # nosec
+
+        prepared_settings = self._prepare_settings_for_request(chat_history, settings)
+        response = await self._async_converse(**prepared_settings)
+
+        return [self._create_chat_message_content(response)]
+
+    @override
+    @trace_streaming_chat_completion(BedrockBase.MODEL_PROVIDER_NAME)
+    async def _inner_get_streaming_chat_message_contents(
+        self,
+        chat_history: "ChatHistory",
+        settings: "PromptExecutionSettings",
+    ) -> AsyncGenerator[list["StreamingChatMessageContent"], Any]:
+        # Not all models support streaming: check if the model supports streaming before proceeding
+        model_info = await self.get_foundation_model_info(self.ai_model_id)
+        if not model_info.get("responseStreamingSupported"):
+            raise ServiceInvalidRequestError(f"The model {self.ai_model_id} does not support streaming.")
+
+        if not isinstance(settings, BedrockChatPromptExecutionSettings):
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, BedrockChatPromptExecutionSettings)  # nosec
+
+        prepared_settings = self._prepare_settings_for_request(chat_history, settings)
+        response_stream = await self._async_converse_streaming(**prepared_settings)
+        for event in response_stream.get("stream"):
```

Comment:
Is there a need to track which choice_index we're on so we know we're combining streaming chunks of the same message and not trying to combine other indices?

Reply:
I have decided to limit the number of responses to one for the Bedrock chat completion interface. This is because of the following:
1. There is only one series of models that support multiple chat completion responses. They are the AI21 Labs Jamba models: https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-jamba.html. However, I was not able to get the model to return multiple responses when I requested it. The content block in theory should contain multiple elements when there are multiple responses but in reality, it always contains a single element: https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_Message.html
2. The AI21 Labs Jamba models cannot stream multiple responses. Also see here: https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-jamba.html